src/c++/vdo/base/dm-vdo-target.c

```diff
@@ -2981,6 +2989,18 @@ static int vdo_preresume_registered(struct dm_target *ti, struct vdo *vdo)
 		vdo_log_info("starting device '%s'", device_name);
 		result = perform_admin_operation(vdo, LOAD_PHASE_START, load_callback,
 						 handle_load_error, "load");
+		if (result == VDO_UNSUPPORTED_VERSION) {
+			/*
+			 * A component version is not supported and the load has been aborted.
+			 */
+			vdo_log_error("Start failed, suspending the VDO");
+			vdo->suspend_type = VDO_ADMIN_STATE_SUSPENDING;
+			perform_admin_operation(vdo, SUSPEND_PHASE_START,
+						suspend_callback, suspend_callback,
+						"suspend");
+			return result;
```

Comment:
You could just return -EINVAL directly here instead of changing the result in the calling function. There's precedent in the clause above (line 2985).

Reply:
I thought that because this has a status code associated with it, it should be handled by addition to the list of statuses translated to -EINVAL in the calling function. The above precedent does not have a status code associated and would thus be less consistent with the existing check in the calling function.